src/Microsoft.SqlTools.ServiceLayer/Rename/RenameUtils.cs

```diff
@@ -0,0 +1,74 @@
+using Microsoft.SqlServer.Management.Common;
+using Microsoft.SqlServer.Management.Sdk.Sfc;
+using Microsoft.SqlServer.Management.Smo;
+using Microsoft.SqlTools.ServiceLayer.Rename.Requests;
+
+namespace Microsoft.SqlTools.ServiceLayer.Rename
+{
+    /// <summary>
+    /// Helper class for Rename Service
+    /// </summary>
+    public static class RenameUtils
+    {
+        private static readonly string UrnPrefix = "Server[@Name='{0}']/Database[@Name='{1}']/Table[@Name='{2}' and @Schema='{3}']";
+        private static readonly string RegexMatchPatternValidObjectName = @"^[\p{L}_][\p{L}\p{N}@$#_]{0,127}$";
+        /// <summary>
+        /// Method to validate all the parameters for the renaming operation
+        /// </summary>
+        /// <param name="requestParams">parameters which should be checked</param>
+        public static void Validate(ProcessRenameEditRequestParams requestParams)
+        {
+            if (requestParams == null)
+            {
+                throw new ArgumentNullException();
+            }
+            if (String.IsNullOrEmpty(requestParams.TableInfo.TableName) || String.IsNullOrEmpty(requestParams.TableInfo.Database) || String.IsNullOrEmpty(requestParams.TableInfo.Schema) || String.IsNullOrEmpty(requestParams.TableInfo.OldName) || String.IsNullOrEmpty(requestParams.ChangeInfo.NewName))
+            {
+                throw new ArgumentException(SR.RenameRequestParametersNotNullOrEmpty);
+            }
+            if (!Regex.IsMatch(requestParams.TableInfo.Schema, RegexMatchPatternValidObjectName) || !Regex.IsMatch(requestParams.TableInfo.Database, RegexMatchPatternValidObjectName) || !Regex.IsMatch(requestParams.TableInfo.OldName, RegexMatchPatternValidObjectName) || !Regex.IsMatch(requestParams.TableInfo.TableName, RegexMatchPatternValidObjectName) || !Regex.IsMatch(requestParams.ChangeInfo.NewName, RegexMatchPatternValidObjectName))
+            {
+                throw new ArgumentOutOfRangeException(SR.NotAllowedCharacterForRenaming);
+            }
+        }
+        /// <summary>
+        /// Method to get the sqlobject, which should be renamed
+        /// </summary>
+        /// <param name="requestParams">parameters which are required for the rename operation</param>
+        /// <param name="connection">the sqlconnection on the server to search for the sqlobject</param>
+        /// <returns>the sqlobject if implements the interface IRenamable, so they can be renamed</returns>
+        public static IRenamable GetSQLRenameObject(ProcessRenameEditRequestParams requestParams, SqlConnection connection)
+        {
+            ServerConnection serverConnection = new ServerConnection(connection);
+            Server server = new Server(serverConnection);
+
+            Database database = new Database(server, requestParams.TableInfo.Database);
+            SqlSmoObject dbObject = database.Parent.GetSmoObject(new Urn(GetURNFromDatabaseSqlObjects(requestParams, serverConnection.TrueName)));
```

Comment:
>etSmoOb [](http://example.com/codeflow?start=53&length=7)

i think GetSmoObject will work if the Urn doesn't have a Name attribute on the Server node.

Reply:
Ok, I will remove it and check it.

Reply:
So I removed the server name Name attribute. If I remove that, I get the following error:
```
Microsoft.SqlServer.Management.Smo.FailedOperationException: Attempt to retrieve data for object failed for Server '127.0.0.1,1433'
Microsoft.SqlServer.Management.Smo.SmoException: For the moment we don't support Urn's that do not start with Server
``` 
The Urn looks fine for me though: ```"Database[@Name='RenameTest_044efd534acc44e489ee03fd00712b42']/Table[@Name='testTable1_RenamingTable' and @Schema='dbo']"```

Reply:
I meant you could just do "Server/Database[@Name=" instead of "Server[@Name='s']/Database" The Server node in the Urn is still required, just not the Name filter. Unless that's what you tried; your comment implies your Urn starts with Database though.